gateway/handle_http_auth_test.go

```diff
@@ -391,4 +406,118 @@ func TestAuth(t *testing.T) {
 			require.Equal(t, "Invalid replay source\n", string(body))
 		})
 	})
+
+	t.Run("authDestIDForSource", func(t *testing.T) {
```

Comment:
@mihir20 You might need to update these tests since some response codes got changed.